### ISSUE TITLE
Unable to install ibm-db on python:3.11.2-slim-buster image on MacOS M1 Chip System, need to find a image that works on building ibm_db

### ISSUE THREAD

**Issue by j-mon** [ibmdb/python-ibmdb#835]
- Operating System Name:
- db2level output from clidriver if in use:
- Target Db2 Server Version:
- Python Version: 3.11
- ibm_db version: ibm_db
- For non-Windows, output of below commands:
  uname: Darwin
  uname -m: arm64
- Value of below environment variables if set:
  IBM_DB_HOME:
  PATH:
  LIB/LD_LIBRARY_PATH/DYLD_LIBRARY_PATH:

- Test script to reproduce the problem.
- For installation related issue, complete output of `pip install ibm_db` command.

Error:
#0 3.188         File "/tmp/pip-build-env-nkbtdim_/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 335, in run_setup
#0 3.188           exec(code, locals())
#0 3.188         File "<string>", line 350, in <module>
#0 3.188       NameError: name 'arch_' is not defined
#0 3.188       [end of output]

Steps to Reproduce:

1. add ibm-db-sa and ibm-db to requirements.txt
2. docker-compose up -d --build

**Comment by Earammak** [comment]
@j-mon, The problem you are pointing for python 3.11 and the error mentioned here is for python 3.9. 
Can you please share the complete output of "pip install ibm_db".

**Comment by bimalkjha** [comment]
@j-mon I hope you are able to install ibm_db using right steps now. Please let us know if we can close this issue? Thanks.

**Comment by bimalkjha** [comment]
export ARCHFLAGS="-arch x86_64"

**Comment by stealthco-tony** [comment]
Can we just "update the pypi package" instead of needing a complete separate homebrew/compiling from source workaround? This isn't meeting users where they are at all; in fact, its the complete opposite. Thanks.

**Comment by bimalkjha** [comment]
@stealthco-tony The latest version of ibm_db do not need compiler as it comes as a precompiled wheel package. But you need x64 version of python to install ibm_db. Thanks.

**Comment by IsraAgus** [comment]
Hi, I followd the steps and this is the output: 
ImportError                               Traceback (most recent call last)
Cell In[8], [line 1](vscode-notebook-cell:?execution_count=8&line=1)
----> [1](vscode-notebook-cell:?execution_count=8&line=1) 

import ibm_db

ImportError: dlopen(/Users/israelvargas/Documents/IBM Proyectos/MATT/Projects/Finanzas/.venv/lib/python3.9/site-packages/ibm_db.cpython-39-darwin.so, 0x0002): Symbol not found: __ZNKSt8__detail20_Prime_rehash_policy14_M_need_rehashEmmm
  Referenced from: <803DFCA0-2956-3B4C-B633-4BCFBAA8B43A> {path_to_local_enviroment}/libdb2.dylib
  Expected in:     <3FE24BF6-7E7F-338E-B009-41FD96FDB569> /usr/lib/libstdc++.6.dylib


**Python version:** 3.9
**gcc -v:**  x86_64-apple-darwin23.6.0
This is a virtual environment created based on 3.9 Python intel.
I think that it is similar to the expected error that you mentioned in the instructions, but I cannot solve it yet. Is there any hint?

**Comment by j-mon** [comment]
It’s impossible to solve it’s a crazy bug let me know how it goes lol


*Jeffrey Monsalve *

Stevens Institute of Technology *|*
2022
Rutgers University *|*
2017 *|*
Mobile: (201)-850-2684 *|*

<http://www.linkedin.com/in/jemonsalve/>

**Comment by j-mon** [comment]
I think it’s missing a file in a path ?


*Jeffrey Monsalve *

Stevens Institute of Technology *|*
2022
Rutgers University *|*
2017 *|*
Mobile: (201)-850-2684 *|*

<http://www.linkedin.com/in/jemonsalve/>

**Comment by bimalkjha** [comment]
@IsraAgus Please share output of below commands to understand your system environment:
```
otool -L /Users/israelvargas/Documents/IBM Proyectos/MATT/Projects/Finanzas/.venv/lib/python3.9/site-packages/ibm_db.cpython-39-darwin.so
gcc -v  => complete output.
echo $DYLD_LIBRARY_PATH
ls /usr/local/Homebrew/Cellar/gcc
```
Also, check troubleshooting instructions as documented here: https://github.com/ibmdb/python-ibmdb/blob/master/INSTALL.md#-4-troubleshooting-post-install-errors Thanks.

**Comment by j-mon** [comment]
This has been an issue for a while please address. It’s an issue for users.
I don’t work there anymore.


*Jeffrey Monsalve *

Stevens Institute of Technology *|*
2022
Rutgers University *|*
2017 *|*
Mobile: (201)-850-2684 *|*

<http://www.linkedin.com/in/jemonsalve/>

**Comment by IsraAgus** [comment]
Hello, 

It works now for me. As you mentioned @bimalkjha, I committed a mistake when I configured the environment variable for the file: **libstdc++.6.dylib**. The simplest solution is in the last tutorial you shared in the troubleshooting section: https://github.com/ibmdb/python-ibmdb/blob/master/INSTALL.md#-4-troubleshooting-post-install-errors 

It is crucial to mention that **when you start the installation and all the stuff, be sure that the zsh prompt is running in the x86 architecture; otherwise, the installation will not work.**

For future readers: 
* Read the instructions carefully.
* Always work in your terminal with the x86 architecture.
